x-pack/test/functional/config.js

```diff
@@ -162,7 +162,6 @@ export default async function ({ readConfigFile }) {
       `--server.uuid=${env.kibana.server.uuid}`,
       `--server.port=${servers.kibana.port}`,
       `--elasticsearch.url=${formatUrl(servers.elasticsearch)}`,
-      '--xpack.monitoring.kibana.collection.enabled=false',
```

Comment:
I had to remove this in order for `const { collectorSet } = server.plugins.monitoring` to work in the API handler. 

Maybe it'd be better to expose the collectorSet regardless, but check for `xpack.monitoring.kibana.collection.enabled` in the bulk fetch method of the collector set ...

Reply:
On second thought, I don't really like the idea of lazy checking a config setting in the bulk method.

The `collectorSet` only gets exposed on the monitoring plugin when `xpack.monitoring.kibana.collection.enabled` is true, and I think that's fine.

Maybe when collection is disabled, we can expose a no-op function on the plugin so other plugins don't have to think about it.